### DISCUSSION TITLE
Best way to update?

### DISCUSSION THREAD

**Discussion by Jiayou-Chao** [pear-devs/pear-desktop#2348] [Q&A] [answered]
Is there a way to automatically update the app?

**Comment by ArjixWasTaken** [comment]
If you installed youtube music using a package manager, then you could use the package manager to update youtube music.
If you did a manual install, no there is no automatic way to update.

**Reply by Jiayou-Chao** [reply]
Thank you.

**Comment by Lockszmith-GH** [comment, marked as answer]
Like ArjixWasTaken mentioned, you can use a package manager.

Me, I'm on Windows 11, and I personally use Microsoft's own WinGet (via UniGetUI), the package name is: `th-ch.YouTubeMusic`

[marticliment/UniGetUI GitHub repo link](https://github.com/marticliment/UniGetUI?tab=readme-ov-file#unigetui-formerly-wingetui)

In the readme there you'll also see links to the WinGet and Scoop package managers, UniGetUI will allow you to manage the installation as well as manage automated updates with a nice UI - so the best of both worlds.

**Reply by dieBakterie** [reply]
UniGetUI seems to have issues, with installing and updating the `th-ch.YouTubeMusic` package, could be the _packagers_ fault tho, just wanna say this instead of opening an issue i guess?

**Reply by ArjixWasTaken** [reply]
UniGetUI is a wrapper around winget and other tools, have you tried directly using winget in the terminal?
Does it succeed?

(trying to narrow down the issue)

**Reply by dieBakterie** [reply]
It's some weird previous version files can't be removed, don't think it's an upstream issue of the package since only i've this issue?

**Reply by dieBakterie** [reply]
Weirdly this is where my current install is located `C:\Users\USERNAME\AppData\Local\Temp\nsa1E65.tmp\old-install`, so maybe it has something todo with the update/download issue that's already open?

**Comment by KaanAlper** [comment]
I made this for debian based distros
https://github.com/KaanAlper/th-ch-Youtube-Music-Auto-Updater-Installer

**Comment by RileyXX** [comment]
I installed with `winget install th-ch.YouTubeMusic`

However, now that a new version is available I try to update with `winget upgrade th-ch.YouTubeMusic` and it says no new versions are available. Not sure what the issue is.

When I used `winget install th-ch.YouTubeMusic --force` it says it's installing v3.7.4 but when I open the app its v3.7.0 and prompts me to update. 

It's installing for me in `C:\Users\USERNAME\AppData\Local\Programs\youtube-music`

Something definitely seems to be broken. Because I am also seeing that v3.7.5 is available now and it is still trying to install v3.7.4. No idea what the issue could be. I'm trying to automate the updating with winget so I can schedule daily update checks with windows task scheduler.

**Reply by ArjixWasTaken** [reply]
The latest version on winget is 3.7.4
![Screenshot_2025-03-06-03-00-02-159_com.github.android.jpg](https://github.com/user-attachments/assets/79efa545-7d75-4524-a54b-68a83a4a6440)

**Reply by ArjixWasTaken** [reply]
Looks like the automatic PR failed.

https://github.com/th-ch/youtube-music/actions/runs/13591273708

@JellyBrick
